redhat-developer/app-services-cli
# feat(kafka): add a terms and conditions check

Closes https://github.com/bf2fc6cc711aee1a0c2a/cli/issues/528

### Description


This PR adds a pre-check before creating a Kafka instance to see if the current user has accepted the terms and conditions - if they have not, print the URL for the user to accept the terms and conditions and exit the program quietly.

Example:
```shell
$ ./rhoas kafka create
In order to create a Kafka instance, you must first review and accept the terms and conditions:

https://www.redhat.com/wapps/tnc/ackrequired?site=ocm&event=register
```

### Verification Steps

1. Create a new Red Hat user account at https://sso.redhat.com/auth/realms/redhat-external/login-actions/registration?client_id=https%3A%2F%2Fwww.redhat.com%2Fwapps%2Fugc-oidc&tab_id=LSRtRa-Vv68
2. Log in to rhoas CLI with your new user.
3. Run `rhoas kafka create <kafka-name>` and you should see the following message:
```
In order to create a Kafka instance, you must first review and accept the terms and conditions:

https://www.redhat.com/wapps/tnc/ackrequired?site=ocm&event=register
```

4. Accept the T&Cs and run the kafka create command - the T&C check will be passed as you have already accepted them.

### Type of change


### Checklist

~- [ ] Documentation added for the feature~
